api/src/main/java/org/openmrs/module/registrationcore/api/impl/IdentifierBuilder.java

```diff
@@ -37,7 +37,7 @@ public void setPatientService(PatientService patientService) {
      * @param location identifier location
      * @return generated patient identifier
      */
-    public PatientIdentifier generateIdentifier(Integer sourceId, Location location) {
+    public PatientIdentifier generateIdentifier(String sourceId, Location location) {
         location = getLocation(location);
```

Comment:
I don't know if there are any implications, but this is changing the method signature of a public method.  Might be safer to add a second method that takes a string and performs the necessary logic.

Reply:
Isn't it safe now, having this more generic. I trust it accepts both String and Integer values @mseaton

Reply:
@samuelmale the problem might be coming from JARs that depend on Idgen and that will complain that this signature is not found anymore:
```java
generateIdentifier(Integer, Location)
```
Those issues can actually be quite nasty to troubleshoot.

As @mseaton said just introduce a new method, keeping the old signature around. In terms of the implementation, don't duplicate code, you will reuse one in the other.